meson.build

```diff
@@ -0,0 +1,142 @@
+geoip_dep           = dependency('geoip', required: get_option('geoip'))
+libproxy_dep        = dependency('libproxy', required: get_option('libproxy'))
+libnotify_dep       = dependency('libnotify', required: get_option('libnotify'))
+libappindicator_dep = dependency('libappindicator', required: get_option('libappindicator'))
+
+trg_deps += [geoip_dep, libproxy_dep, libnotify_dep, libappindicator_dep]
+
+# compiler
+cc = meson.get_compiler('c')
+
+# compiler deps
+trg_deps += cc.find_library('m', required: false)
+
+# nl_langinfo check
+nl_langinfo = cc.has_function('nl_langinfo', prefix: '#include <langinfo.h>')
+nl_li_feature = get_option('nl_langinfo')
+if nl_li_feature.enabled() and not nl_langinfo
+  error('nl_langinfo() specified but not found')
+endif
+nl_langinfo = nl_langinfo and nl_li_feature.allowed()
+
+
+# config data
+conf_data = configuration_data()
+
+conf_data.set_quoted('PACKAGE_NAME', project_name)
+conf_data.set_quoted('PACKAGE_VERSION', project_version)
+conf_data.set_quoted('GETTEXT_PACKAGE', project_name)
+conf_data.set_quoted('LOCALEDIR', get_option('localedir'))
+
+# debug options
+debug = get_option('debug')
+conf_data.set('DEBUG', debug)
+conf_data.set('G_ENABLE_DEBUG', debug)
+conf_data.set('G_DISABLE_ASSERT', not debug)
+conf_data.set('G_DISABLE_CHECKS', not debug)
+conf_data.set('G_DISABLE_CAST_CHECKS', not debug)
+
+# glib API configuration
+maj_min = glib_max_version.split('.')
+conf_data.set('GLIB_VERSION_MAX_ALLOWED',
+              'GLIB_VERSION_@0@_@1@'.format(maj_min[0], maj_min[1]))
+maj_min = glib_min_version.split('.')
+conf_data.set('GLIB_VERSION_MIN_REQUIRED',
+              'GLIB_VERSION_@0@_@1@'.format(maj_min[0], maj_min[1]))
+
+# compile time features
+conf_data.set('HAVE_RSS', rss_dep.found())
+# TODO: why??
+conf_data.set10('HAVE_GEOIP', geoip_dep.found())
+conf_data.set('HAVE_LIBPROXY', libproxy_dep.found())
+conf_data.set('HAVE_LIBNOTIFY', libnotify_dep.found())
+conf_data.set('ENABLE_NL_LANGINFO', nl_langinfo)
+conf_data.set('HAVE_LIBAPPINDICATOR', libappindicator_dep.found())
+
+# compiler/linker flags
+flags = ['-funsigned-char',
+         '-fstack-protector-strong',
+         '-Winline',
+         '-Wno-padded',
+         '-Wno-unused-parameter',
+         '-Wstrict-prototypes',
+         '-Wmissing-prototypes',
+         '-Werror=format=2',
+         '-Werror=implicit-function-declaration',
+         '-Werror=pointer-arith',
+         '-Werror=format-security',
+         '-Werror=init-self',
+         '-Werror=missing-include-dirs',
+         '-Werror=date-time',
+         '-Wno-conversion',
+         '-Wno-sign-compare',
+         '-Wno-deprecated-declarations',
+         '-Wno-deprecated',
+         '-Wno-error=format-nonliteral']
+
+cflags = cc.get_supported_arguments(flags)
+add_project_arguments(cflags, language: 'c')
+add_project_arguments('-DHAVE_CONFIG_H', language: 'c')
+
+# link arguments
+ldflags = cc.get_supported_link_arguments('-pie',
```

Comment:
`-pie` is questionable. `executable()` and other build targets take a `pie: true` argument.